### DISCUSSION TITLE
What's the difference?

### DISCUSSION THREAD

**Discussion by tahamansoor** [aws/aws-sdk-js-v3#5634] [Q&A] [answered]
mostly every service client in aws-sdk v3 can be imported/used two ways
for example dynamodb
first:
`import { DynamoDBClient } from '@aws-sdk/client-dynamodb'`
second:
`import { DynamoDB } from '@aws-sdk/client-dynamodb'`

what's the difference

what i know is little difference like 

the second one can be used like v2 except for some difference

**Comment by RanVaknin** [comment, marked as answer]
Hi @tahamansoor ,

The main differences between the two import methods is syntax and structure. Both achieve similar outcomes, but they cater to different coding styles.

When you use:
```js
import { DynamoDBClient } from '@aws-sdk/client-dynamodb';
```
You're taking advantage of JS SDK v3's modular approach. Here, you directly import the client and then utilize it to send commands. For instance:
```js
import { DynamoDBClient, PutItemCommand} from '@aws-sdk/client-dynamodb'; // adding here each command you want to use.

const client = new DynamoDBClient(/* config */);
client.send(new PutItemCommand(/* command parameters */));
```

On the other hand, the following import:

```js
import { DynamoDB } from '@aws-sdk/client-dynamodb';
```
is more closely aligned with the v2 SDK's coding style, offering a familiar interface for those who are transitioning from v2. In this case, you would use the service class directly, like so:

```js
import { DynamoDB } from '@aws-sdk/client-dynamodb';

const dynamoDB = new DynamoDB(/* config */);
dynamoDB.putItem(/* params */);
```

Hope this clarifies the differences.
Thanks,
Ran~

**Reply by tahamansoor** [reply]
I have another question.
I'll ask on another discussion

**Reply by rdsedmundo** [reply]
I think this answer should also highlight this:
> If you use tree shaking to reduce bundle size, using non-modular interface will increase the bundle size as compared to using modular interface.

As seen in the README.
